dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -180,6 +186,138 @@ def dpnp_add(x1, x2, out=None, order="K"):
     return dpnp_array._create_from_usm_ndarray(res_usm)
 
 
+_bitwise_and_docstring_ = """
+bitwise_and(x1, x2, out=None, order='K')
+
+Computes the bitwise AND of the underlying binary representation of each
+element `x1_i` of the input array `x1` with the respective element `x2_i`
+of the input array `x2`.
+
+Args:
+    x1 (dpnp.ndarray):
+        First input array, expected to have integer or boolean data type.
+    x2 (dpnp.ndarray):
+        Second input array, also expected to have integer or boolean data
+        type.
+    out ({None, dpnp.ndarray}, optional):
+        Output array to populate.
+        Array have the correct shape and the expected data type.
+    order ("C","F","A","K", optional):
+        Memory layout of the newly output array, if parameter `out` is `None`.
+        Default: "K".
+Returns:
+    dpnp.ndarray:
+        An array containing the element-wise results. The data type
+        of the returned array is determined by the Type Promotion Rules.
+"""
+
+
+def dpnp_bitwise_and(x1, x2, out=None, order="K"):
+    """Invokes bitwise_and() from dpctl.tensor implementation for bitwise_and() function."""
+
+    # dpctl.tensor only works with usm_ndarray or scalar
+    x1_usm_or_scalar = dpnp.get_usm_ndarray_or_scalar(x1)
+    x2_usm_or_scalar = dpnp.get_usm_ndarray_or_scalar(x2)
+    out_usm = None if out is None else dpnp.get_usm_ndarray(out)
+
+    func = BinaryElementwiseFunc(
+        "bitwise_and",
+        ti._bitwise_and_result_type,
+        ti._bitwise_and,
+        _bitwise_and_docstring_,
+    )
```

Comment:
Why are we paying the cost of creating `BinaryElementwiseFunc` instance at every invocation? Create it during module import and use here instead.